src/main/java/org/opentripplanner/middleware/controllers/api/UserController.java

```diff
@@ -37,6 +39,17 @@ User preCreateHook(User user, Request req) {
 
     @Override
     User preUpdateHook(User user, User preExistingUser, Request req) {
+        // In order to update a user, the updating user must be authenticated. Because this API is registered under the
+        // "public" path, this is not done before the request (see Main class where routes are registered).
+        Auth0Connection.checkUser(req);
+        Auth0UserProfile requestingUser = Auth0Connection.getUserFromRequest(req);
+        // Additionally, if the user is attempting to update someone else's profile, they must be an admin.
+        if (!user.auth0UserId.equals(requestingUser.user_id)) {
```

Comment:
This logic needs to be used in most hooks of other controllers. Also in the preDeleteHook of this controller.